### PR TITLE
Refactor _S to modern CSS layout standards using grid and flex

### PR DESCRIPTION
**Pull request by mor10**
#### Changes proposed in this Pull Request:
This pull request refactors _S to use CSS grid and flex as main layout modules:
- Simplified semantic HTML structure by removing containers used for float-based layout nesting.
- Two-column grid layout using `grid-template-areas` in content-sidebar.css and sidebar-content.css.
- Main navigation and Comments, Posts, and Post navigation layout refactored to use flex.
- style.css and related Sass files updated.

The refactoring in this pull request prepares _S for modern layouts. Each main component (header, main, sidebar, footer) is a direct descendant of the site grid and can be placed anywhere within that grid. Flex enables fluid responsive layouts for the main navigation and comments, posts, and post navigation. 

Flex and grid both honor text direction meaning this pull request makes _S fully RTL compatible out of the box without the need for an additional stylesheet.

Note: Since there are extensive edits in this pull request, I've intentionally _not_ quashed the individual commits to make it easier to see exactly what has been done and why. 

#### Related issue(s):
Main issue: #1250
Compatible with #1246

### CODE REVIEW

File: header.php [hunk 1]
```diff
@@ -16,6 +16,7 @@
 	<meta charset="<?php bloginfo( 'charset' ); ?>">
 	<meta name="viewport" content="width=device-width, initial-scale=1">
 	<link rel="profile" href="http://gmpg.org/xfn/11">
+	<link rel="stylesheet" id="_s-style-css" href="http://underscores.local:8888/wp-content/themes/underscores/layouts/content-sidebar.css" type="text/css" media="all">
```
Comment (slushman) [inline header.php]: This shouldn't be here, right?

**Comment by mor10** [comment]
No, it should not. I had it in to run a test and forgot to remove it. Will update pronto.

**Comment by mor10** [comment]
FYI I have removed the errant test data. The PR should now be correct.

**Comment by obenland** [comment]
I’ve not tested it but I like the idea. 

Looks like Travis found some white-space issues. Could we replace the `#secondary` selector with a non-id one?

**Comment by mor10** [comment]
@obenland I'll sort out the whitespace issues. Considering there is just one widgetized area in the theme, I could just remove the `#secondary` selector, but I'll see if I can come up with another way of just targeting the sidebar widgets, possibly buy using the sidebar container class. I'll update later today.

**Comment by samikeijonen** [comment]
Skip link should go to `#main` now in `header.php`? And also change the CSS part of it.

**Comment by mor10** [comment]
Ummm... Looks like Travis had some sort of minor meltdown. Any way to force it to run again? Or am I mistaken?

**Comment by mor10** [comment]
All checks have passed. What are your thoughts?

**Comment by alaczek** [comment]
@mor10 this is looking good! Are you able to resolve the conflicts? If not I can help.
I saw you comment above about individual commits, but since these changes have been sitting here for a while now, could you squash too?

**Comment by mor10** [comment]
I'll look at it tomorrow

**Comment by Ismail-elkorchi** [comment]
@mor10 Have you had a chance to look at this yet?

**Comment by jaclyntan** [comment]
Is this PR being merged anytime soon?

**Comment by crunnells** [comment]
There are some merge conflicts, mostly white space related, but I'd like to evaluate and see if we can get this merged soon.

**Comment by mor10** [comment]
I'll give it another pass tomorrow

**Comment by PeterLorre** [comment]
Bump

**Comment by Ismail-elkorchi** [comment]
Hi everyone ! I've rebased this pull request on master, resolved the merge conflicts, fixed some 
WPCS issues, added some tweaks and squashed the commits.

@mor10 Would you like to check the changes ?

@crunnells Do you still want to take a look a look at this PR ?

**Pull request merged**